### ISSUE TITLE
statement is not reset between each execution

### ISSUE THREAD

**Issue by sluggern67** [markdirish/node-odbc#157]
**Describe your system**
* `odbc` Package Version: 2.3.5
* ODBC Driver: unixODBC
* Database Name: IBM i Access ODBC Driver
* Database Version: DB2 for i
* Database OS: V7R3
* Node.js Version: 14.11.0
* Node.js OS: IBM i

**Describe the bug**
The result of a statement.execute() is accumulated with the previous execute.

**Expected behavior**
It should only containt the result of the last execution.

**To Reproduce**
Run code below.
I would expect the output in the log to be
4
4
4

It actually returns
4
8
12

**Code**
const odbc = require('odbc');

async function test(){
    const connection = await odbc.connect('DSN=NODE.JS_WEBSERVICES;PWD=XXXXXXXX;');
    await connection.query(`CREATE OR REPLACE TABLE testlib.testfile(recordId int, code varchar(1)) ON REPLACE DELETE ROWS`);
    await connection.query(`INSERT INTO testlib.testfile VALUES(1, 'A'),(2, 'A'),(3,'A'),(4,'A')`);
    const stmt = await connection.createStatement();
    await stmt.prepare(`SELECT * FROM testlib.testfile WHERE code=?`);
    for(i=0; i<3; i++){
        await stmt.bind(['A']);
        let result = await stmt.execute();
        console.log(result.length);
    }
}

test();

**Comment by markdirish** [comment]
Good find. I've been implementing a cursor functionality to allow `fetch` calls and I noticed something similar, but I didn't realize it affected statements as well.

In this commit, I added some logic to delete the underlying bound data after it has all been converted to JavaScript values: https://github.com/markdirish/node-odbc/pull/152/commits/7aa138c7b542522ffc2b484780f83c82a7f1ac56 This should remedy both issues. Will close issue when I get it merged in.
